### ISSUE TITLE
Knowledge - Ingesting of files for the same website(already ingested) is very fast.

### ISSUE THREAD

**Issue by sangee2004** [otto8-ai/otto8#722]
**Steps to reproduce the problem:**
1.  Create an agent with knowledge from any website say - `acorn.io`
2. Wait for all files to be synced and then ingest all files. In my case , ingestion took 6 -7 minutes to complete . I saw few ingestion failures that succeed on retry.
3. Once ingestion of all the files is completed, delete this knowledge.
4. Add the same knowledge website - `acorn.io` to this agent.

Notice that ingestion happens very fast.  Ingestion of all files is almost immediate.

https://github.com/user-attachments/assets/7e1e8b00-7786-4fe9-aa71-182c4b39f4f1

5. The same behavior is also noted when creating a new agent and adding the same  website `acorn.io`.

Ingestion of files happen very fast.

https://github.com/user-attachments/assets/03b5717f-a3a7-4e54-bbaf-d2c249a3c403

**Question:**
I am seeing ingestion being attempt for each file even for step3 and step 5. 
But i am not sure how they get to ingested state so quickly. Is there some caching that is causing these ingestion calls to succeed faster the second time?

**Comment by iwilltry42** [comment]
Coming back to this today:
Do we delete the data from the DB when I delete the knowledge?
We did, when a file was "unapproved" or when the last knowledge source was removed from an agent. Until today, we did not (due to a bug) delete a file when removing one knowledge source from an agent if it was not the last one.

add acorn.io again. It takes no time to scrape and sync. Is that by design?
This is not related to anything we do - in fact, those are two completely separate scrape- and ingestion processes with no shared cache whatsoever (I double checked that).
However, there are two large time-consuming steps during ingestion:

1. website-cleaning (LLM call)
2. (if needed, for PDF/PPTX/DOCX) VLM call for picture interpretation

Both are calls to the OpenAI chat-completion API and seem to get cached (definitely the former).
Here are example timings just for the website-cleaning:

First run:
```
Cleaned website content in 901.727258ms
Cleaned website content in 1.012397428s
Cleaned website content in 809.27944ms
Cleaned website content in 832.195671ms
Cleaned website content in 789.839448ms
Cleaned website content in 1.119338769s
```

Second run:
```
Cleaned website content in 46.787958ms
Cleaned website content in 24.958816ms
Cleaned website content in 45.278302ms
Cleaned website content in 41.407286ms
Cleaned website content in 30.614086ms
Cleaned website content in 29.907859ms
```

Caching is not active for the embeddings model (according to the docs): https://platform.openai.com/docs/guides/prompt-caching
